internal/pkg/util/yamlprotoconverter_test.go

```diff
@@ -0,0 +1,214 @@
+			`
+and_result:
+  result_predicates:
+    - resource_names_fragment:
+        field: 1
+        element: 0
+        action:
+          regex_action:
+            pattern: "some_regex"
+            replace: "a_replacement"
+    - request_node_fragment:
+        field: 2
+        action:
+          regex_action:
+            pattern: "some_regex_for_node_fragment"
+            replace: "another_replacement"
+`,
+			&ResultPredicate{},
+			&ResultPredicate{
+				Type: &aggregationv1.ResultPredicate_AndResult_{
+					AndResult: &AndResult{
+						ResultPredicates: []*ResultPredicate{
+							{
+								Type: &aggregationv1.ResultPredicate_ResourceNamesFragment_{
+									ResourceNamesFragment: &ResourceNamesFragment{
+										Field:   1,
+										Element: 0,
+										Action: &ResultAction{
+											Action: &aggregationv1.ResultPredicate_ResultAction_RegexAction_{
+												RegexAction: &RegexAction{
+													Pattern: "some_regex",
+													Replace: "a_replacement",
+												},
+											},
+										},
+									},
+								},
+							},
+							{
+								Type: &aggregationv1.ResultPredicate_RequestNodeFragment_{
+									RequestNodeFragment: &RequestNodeFragment{
+										Field: 2,
+										Action: &ResultAction{
+											Action: &aggregationv1.ResultPredicate_ResultAction_RegexAction_{
+												RegexAction: &RegexAction{
+													Pattern: "some_regex_for_node_fragment",
+													Replace: "another_replacement",
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	},
+	{
+		Description: "KeyerConfiguration + RequestTypeMatch + StringFragment",
+		Parameters: []interface{}{
+			`
+fragments:
+- rules:
+  - match:
+      request_type_match:
+        types:
+        - type.googleapis.com/envoy.api.v2.Endpoint
+        - type.googleapis.com/envoy.api.v2.Listener
+    result:
+      string_fragment: "abc"
+`,
+			&KeyerConfiguration{},
+			&KeyerConfiguration{
+				Fragments: []*Fragment{
+					{
+						Rules: []*FragmentRule{
+							{
+								Match: &MatchPredicate{
+									Type: &aggregationv1.MatchPredicate_RequestTypeMatch_{
+										RequestTypeMatch: &RequestTypeMatch{
+											Types: []string{
+												"type.googleapis.com/envoy.api.v2.Endpoint",
+												"type.googleapis.com/envoy.api.v2.Listener",
+											},
+										},
+									},
+								},
+								Result: &ResultPredicate{
+									Type: &StringFragment{
+										StringFragment: "abc",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	},
+}
+
+var _ = Describe("Yamlprotoconverter", func() {
+	DescribeTable("should be able to convert from yaml to proto",
+		func(yml string, protoToUnmarshal proto.Message, expectedProto proto.Message) {
+			err := FromYAMLToProto(yml, protoToUnmarshal)
+			Expect(err).To(BeNil())
+			Expect(proto.Equal(protoToUnmarshal, expectedProto)).To(Equal(true))
+		},
+		positiveTests...)
+})
```

Comment:
maybe add a negative test too?

Reply:
sure, it's in the TODO in the description. I'm going to add a couple of negative tests too. (I have to say I'm really enjoying this BDD-style of testing...)